src/mechanics/mesh/MeshFem.cpp

```diff
@@ -99,7 +99,18 @@ Group<NodeSimple> MeshFem::NodesTotal(DofType d)
 {
     Group<NodeSimple> group;
     for (auto& element : this->Elements)
-        for (int iNode = 0; iNode < element.DofElement(d).Interpolation().GetNumNodes(); ++iNode)
+    {
+        const ElementFem* dofElm;
+        try
+        {
+            dofElm = &(element.DofElement(d));
```

Comment:
Making try/catch blocks part of common logic is bad practice IMO. I suggest adding a `ElementColleciton::Has(DofType)`. That will also be much clearer and there is no need to have this `dofElm` pointer.
~~~cpp
if(not element.Has(d))
   continue;
...
~~~

Another option is to avoid this code here and throw an exception as before. As far as I can tell, you copied the code to `MeshFemDofConvert::NodePointsTotal` anyways. :yum: 
Then, my comment above would apply there.

Reply:
The underlying container is a DoFContainer and this one uses a map to store its data. The public available accessors all use map.at() to get the data and I thought this implies that catching exceptions is the way to go. If I have to use find (Has(d) ) to know if the element is there anyway then it might makes sense to access the items also using find - but this is not implemented. Why?

Reply:
Catching exceptions should be an exception... Why is `ElementCollection::Has(d)` not implemented? For the same reason as everything else: YAGNI. And now that we needed, we could add it.

Reply:
Öhm ok. I meant Find, but understood.